src/ReflectionFunction.php

```diff
@@ -18,25 +18,46 @@
 /**
  * AST-based reflection for function
  */
-class ReflectionFunction extends BaseReflectionFunction
+class ReflectionFunction extends BaseReflectionFunction implements IReflection
 {
     use ReflectionFunctionLikeTrait, InternalPropertiesEmulationTrait;
 
     /**
+     * Name of the function
+     *
+     * @var string|\Closure
+     */
+    private $functionName;
+
+    /**
      * Initializes reflection instance for given AST-node
      *
      * @param string|\Closure $functionName The name of the function to reflect or a closure.
      * @param Function_|null  $functionNode Function node AST
      */
-    public function __construct($functionName, Function_ $functionNode)
+    public function __construct($functionName, Function_ $functionNode = null)
```

Comment:
1. When the `$functionNode` would be missing?
2. Why `$functionNode` wasn't missing before?

Reply:
`$functionNode` will be missing when `$functionName` names a built in function. (When there is no PHP source code to parse.)

In the future, I expect the `ReflectionEngine` to remember where it found named functions, but not in this PR.